### PR TITLE
feat!: added file download tracking (#260)

### PR DESCRIPTION
**Pull request by jpaten**
### Changes
- Added `file_downloaded` event and relevant models as specified in #260
- Added call to this event in `AzulFileDownload`, adding new optional props as needed

### CODE REVIEW

**Review by MillenniumFalconMechanic — CHANGES_REQUESTED** [review]
Looks good, thanks Jonah! I've added a couple of comments. Can you also double-check the GTM parameter value limit? I was under the impression that parameter names are limited to 40 but parameter values can be 100.

**Comment by jpaten** [comment]
@MillenniumFalconMechanic my bad, I read the wrong limit, it is 100 characters. Some file names come very close to that, but I couldn't find any that have more than 100 characters, so this approach is probably fine

File: src/components/Index/components/AzulFileDownload/azulFileDownload.tsx [hunk 1]
```diff
@@ -12,10 +14,16 @@ import {
 } from "./common/constants";
 
 export interface AzulFileDownloadProps {
+  entityName?: string; // The name of the file downloaded.
+  relatedEntityId?: string[]; // An array of IDs of the file's datasets / projects
+  relatedEntityName?: string[]; // An array of names of the file's datasets / projects
```
[the remark this answers is not in the document]
Comment (jpaten) [reply src/components/Index/components/AzulFileDownload/azulFileDownload.tsx:19]: Sounds good, I'll make that change

**Comment by jpaten** [comment]
TODO: writeup breaking change and tag Fran

**Comment by jpaten** [comment]
#### Breaking Changes @frano-m 
- AzulFileDownloaded has the following new required props
  - `entityName`
    - string
    - The filename for the file being downloaded
  - `relatedEntityId`
    - string
    - The ID(s) of the dataset(s) or project(s) associated with the file
    - If there are multiple different associated datasets or projects, they should be stringified in the `viewModelBuilder` before being passed here
  - `relatedEntityName`
    - string
    - The name(s) of the dataset(s) or project(s) associated with the file
    - If there are multiple different associated datasets or projects, they should be stringified in the `viewModelBuilder` before being passed here

File: tests/azulFileDownload.test.tsx [hunk 2]
```diff
@@ -21,6 +21,11 @@ describe("AzulFileDownload", () => {
   const FILE_URL = "https://example.com/storage/file";
   const MOCK_RUN = jest.fn();
   const URL = "https://example.com/repository/file";
+  const TRACKING_PARAMETERS = {
```
Comment (frano-m) [inline tests/azulFileDownload.test.tsx:24]: @jpaten nice!

**Review by frano-m — APPROVED** [review]
LGTM! Thank you @jpaten 🚀

**Review by MillenniumFalconMechanic — APPROVED** [review]
LGTM, thanks Jonah!

File: tests/azulFileDownload.test.tsx (hunk 2 above)
Comment (MillenniumFalconMechanic) [reply tests/azulFileDownload.test.tsx:24]: Agreed!

**Pull request merged**